pkg/kubecost/allocation.go

```diff
@@ -1988,6 +1989,36 @@ func (asr *AllocationSetRange) Accumulate() (*AllocationSet, error) {
 	return allocSet, nil
 }
 
+// NewAccumulation clones the first available AllocationSet to use as the data structure to
+// accumulate the remaining data. This leaves the original AllocationSetRange intact.
+func (asr *AllocationSetRange) NewAccumulation() (*AllocationSet, error) {
```

Comment:
Now that I've thought about it some more, is there any reason this shouldn't be the logic for the standard `Accumulate()`?

Reply:
So, for regular Allocations, from my testing, `NewAccumulation()` almost works exactly like `Accumulate()`, but this is not the case for `AssetSetRange` or `SummaryAllocationSetRange`. For API consistency, the `New___` variation seemed more explicit in that it created copies of the inner data before squashing. I _did_ explore updating `AllocationSetRange::Accumulate()` to match `AssetSetRange` and `SummaryAllocationSetRange`, but there are a large handful of tests which fail after that change, which seems to indicate an expected behavior there that's different from the latter two. 

For the most part, `Accumulate()` is fine to run given that the final SetRange object won't be used post accumulation. However, in the case with caching, we want the raw SetRange to remain immutable, so we need an immutable method _if_ accumulate=true.

Reply:
Sounds perfect, thanks for the thorough response. `New___` is perfect here, I'm a big fan.